### PR TITLE
add debug logs in oauth2 proxy

### PR DESCRIPTION
**Pull request by ramaraochavali**
Commit Message: add debug logs in oauth2 proxy
Additional Description: add debug logs in oauth2 proxy to aid in debugging
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Review by kyessenov — APPROVED** [review]

File: source/extensions/filters/http/oauth2/filter.cc [hunk 1]
```diff
@@ -369,15 +375,18 @@ bool OAuth2Filter::canSkipOAuth(Http::RequestHeaderMap& headers) const {
     if (config_->forwardBearerToken() && !validator_->token().empty()) {
       setBearerToken(headers, validator_->token());
     }
+    ENVOY_LOG(debug, "skipping oauth flow due to valid hmac cookie: \n{}", headers);
```
Comment (kyessenov) [inline source/extensions/filters/http/oauth2/filter.cc:378]: Don't think it's a good idea to print all headers. It probably has secrets and would be quite noisy.

Comment (ramaraochavali) [reply source/extensions/filters/http/oauth2/filter.cc]: I have added this as we print in encode headers already. Any way, removed it now.

**Comment by alyssawilk** [comment]
@kyessenov ping?

File: source/extensions/filters/http/oauth2/filter.cc [hunk 2]
```diff
@@ -369,15 +375,18 @@ bool OAuth2Filter::canSkipOAuth(Http::RequestHeaderMap& headers) const {
     if (config_->forwardBearerToken() && !validator_->token().empty()) {
       setBearerToken(headers, validator_->token());
     }
+    ENVOY_LOG(debug, "skipping oauth flow due to valid hmac cookie}");
```
[the remark this answers is not in the document]
Comment (ramaraochavali) [reply source/extensions/filters/http/oauth2/filter.cc:378]: done

**Review by kyessenov — APPROVED** [review]

**Pull request merged**